src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -278,14 +278,15 @@ async def sink_buffer(sample: Sample[Quantity]) -> None:
             asyncio.create_task(self._resampler.resample(), name="resample")
         )
 
-    def __len__(self) -> int:
+    @property
```

Comment:
I think this shouldn't be a property, as it is doing (underneath) a non-trivial calculation.

```suggestion
```